Framework/Algorithms/src/MuonGroupDetectors.cpp

```diff
@@ -97,14 +97,15 @@ void MuonGroupDetectors::exec() {
     // We will be setting them anew
     outWS->getSpectrum(groupIndex).clearDetectorIDs();
 
+    auto &Y = outWS->mutableY(groupIndex);
+    auto &E = outWS->mutableE(groupIndex);
+
     for (auto &wsIndex : wsIndices) {
       for (size_t i = 0; i < inWS->blocksize(); ++i) {
         // Sum the y values
-        outWS->dataY(groupIndex)[i] += inWS->dataY(wsIndex)[i];
-
+        Y[i] += inWS->y(wsIndex)[i];
         // Sum the errors in quadrature
-        outWS->dataE(groupIndex)[i] = sqrt(pow(outWS->dataE(groupIndex)[i], 2) +
-                                           pow(inWS->dataE(wsIndex)[i], 2));
+        E[i] = sqrt(pow(E[i], 2) + pow(inWS->e(wsIndex)[i], 2));
```

Comment:
Don't do this by hand. If you can figure out whether the workspaces store counts or frequencies, you can use, e.g., `countVariances()` to do the squaring for you, and the respective setter will to the `sqrt`.